src/poker/game/GamePlay.java

```diff
@@ -0,0 +1,362 @@
+    public HandsAndBoard result;
+    public Player player1;
+    public Player player2;
+    GameState gameState;
+    List<Player> playerList;
+    public EquityEvaluator equityEvaluator;
+    boolean firstDecisionSmallBlind = true;
+    boolean firstDecisionPostFlop = true;
+    boolean playerFold = false;
+    public GameResult gameResult;
+
+
+    public GamePlay() {
+        this.deck = new Deck();
+        this.result = deck.shuffleDeckAndGetHandsAndBoard();
+        this.player1 = new Player(100, 0, 0, 0, result.getHand1(), "player1");
+        this.player2 = new Player(100, 0, 0, 0, result.getHand2(), "player2");
+        this.gameState = new GameState(GamePhase.PREFLOP, 0);
+        player1.setHandPlayer(result.getHand1());
+        player2.setHandPlayer(result.getHand2());
+        playerList = new ArrayList<>();
+        playerList.add(player1);
+        playerList.add(player2);
+        this.equityEvaluator = new EquityEvaluator(deck, result.getHand1(), result.getHand2(), result.getBoard());
+        this.gameResult = new GameResult();
+    }
+
+    public void gameStart() {
+        Player nonActivePlayer = player1;
+        Player activePlayer = player2;
+        Player smallBlindPlayer = activePlayer;
+        Player bigBlindPlayer = nonActivePlayer;
+        boolean OutOfPostionPlayer = false;
+        while (playerList.size() > 1) {
+
+            topUpPlayer(nonActivePlayer, activePlayer);
+            preFlopDetails(nonActivePlayer, activePlayer);
+            PossibleAction lastAction = null;
+            for (int i = 0; i < 4; i++) {
+
+                while (nextTurnPlayer(lastAction, OutOfPostionPlayer) && !playerFold) {
+                    if (gameState.gamePhase != GamePhase.PREFLOP && smallBlindPlayer == activePlayer && firstDecisionPostFlop) {
+                        Player temp = activePlayer;
+                        activePlayer = nonActivePlayer;
+                        nonActivePlayer = temp;
+                    }
+                    if (activePlayer != smallBlindPlayer && gameState.gamePhase != GamePhase.PREFLOP) {
+                        OutOfPostionPlayer = true;
+                    } else {
+                        OutOfPostionPlayer = false;
+                    }
+
+                    firstDecisionPostFlop = false;//do wyjebania
+                    lastAction = getAction(lastAction, activePlayer, nonActivePlayer);
+                    handleAction(lastAction, activePlayer, nonActivePlayer);
+                    Player temp = activePlayer;
+                    activePlayer = nonActivePlayer;
+                    nonActivePlayer = temp;
+                    if (activePlayer.getStackPlayer() == 0 || nonActivePlayer.getStackPlayer() == 0) {
+                        gameState.setGamePhase(GamePhase.RIVER);
+                        i = 3;
+                    }
+                    if (lastAction == PossibleAction.Fold) {
+                        playerFold = true;
+                    }
```

Comment:
to powinno byc w hadleAction JAKO ZE TO MIALA BYC FUNKCJA DO OBLSUGI AKCJI I WSZYSTKEIGO CO Z NIA ZWIAZANE a wykrycie czy jest to allin jak najbardziej sie w to wpisuje